src/Glpi/Kernel/Kernel.php

```diff
@@ -52,7 +52,7 @@ final class Kernel extends BaseKernel
 
     public function __construct(?string $env = null)
     {
-        $env ??= $_ENV['GLPI_ENVIRONMENT_TYPE'] ?? $_SERVER['GLPI_ENVIRONMENT_TYPE'] ?? 'production';
+        $env ??= \defined('GLPI_ENVIRONMENT_TYPE') ? \GLPI_ENVIRONMENT_TYPE : ($_ENV['GLPI_ENVIRONMENT_TYPE'] ?? $_SERVER['GLPI_ENVIRONMENT_TYPE'] ?? 'production');
```

Comment:
IMHO, it should fail if this constant is not defined. It may permit to detect issues in the autoloading logic.

```suggestion
        $env = GLPI_ENVIRONMENT_TYPE;
```